runtime/help/options.md

```diff
@@ -448,6 +448,8 @@ or disable them:
 * `diff`: integrates the `diffgutter` option with Git. If you are in a Git
    directory, the diff gutter will show changes with respect to the most
    recent Git commit rather than the diff since opening the file.
+* `formatter`: Formatter settings can be for any type of file (javascript,
+   go, python).
```

Comment:
Do we really need this option...activating the plugin will already provide this functionality or am I wrong?
Maybe I oversee something in the code, because I couldn't find a reference. The option which is registered at runtime is `formatter.makeCommands`.

Reply:
The `formatter.makeCommands` option defines whether formatters can be called in the command bar by `name` or the first string of the formatter's `cmd`.

For example, if I have the `"formatter.makeCommands": true` option and the following formatter:

```lua
function init()
    formatter.setup({
        { cmd = 'goimports -w %f', filetypes = { 'go' } },
    })
end
```

I can format the file by opening the command bar (`Ctrl-e`) and executing the `> goimports` command (if the formatter does not have a `name` the first `cmd` string will be used to create the cmmando).

Or I could define a name to be called:

```lua
function init()
    formatter.setup({
        { cmd = 'goimports -w %f', name = 'goimp', filetypes = { 'go' } },
    })
end
```

So the formatter can be applied with the `> goimp` command.
 
The only scenario in which a formatting command will be created with the `formatter.makeCommands` option disabled is when you define bindings to run the formatter, as in the example below:

```lua
function init()
    formatter.setup({
        { cmd = 'goimports -w %f', bind = 'Alt-i', filetypes = { 'go' } },
    })
end
```

In this case a command needs to be created to be able to link with `Alt-i`.

These commands are in case you want to run a single formatter.
To run all formatters of a file use the `> format` command that the formatter plugin provides.

The `formatter.makeCommands` option allows the user to explicitly say when he wants all formatters to become commands.

Reply:
Regarding the `formatter` option, it is used to deactivate the plugin, for example, if I wanted to disable the `ftoptions` plugin that is installed by default I would have to add `"ftoptions": false` to my settings. This is also true for third-party plugins, for example `"lsp": false` will disable the `lsp` plugin if it is installed.

Reply:
> Regarding the `formatter` option, it is used to deactivate the plugin, for example

I totally forget about [them](https://github.com/zyedidia/micro/blob/master/internal/config/plugin.go#L104), so it was a stupid question. :man_facepalming: 
But the basic idea of this plugin is very good and will definitely become handy.

You even though on `ReOpen()` already. :+1: